### PR TITLE
Export ComponentList, ComponentListIterator.

### LINKED ISSUE #360: ComponentList and ComponentListIterator are not exported.
Should they be? I'll do this quickly if we agree it should be done.

### PR DESCRIPTION
**Pull request by chrisdembia**
Fixes #360.

### CODE REVIEW

**Comment by chrisdembia** [comment]
Appveyor was failing. To fix that, I chose to put ComponentListIterator into its own header. I'd like to request @aymanhab and @aseth1 as reviewers? Is it okay that I put some additional includes into `osimCommon.h`? I wasn't sure if adding those includes is okay, since a bunch are missing from that file.

File: OpenSim/Common/ComponentList.h [hunk 1]
```diff
@@ -26,14 +26,13 @@
...
-#include "OpenSim/Common/Object.h"
-#include "Simbody.h"
+#include <Simbody.h>
 
 namespace OpenSim {
 
 class Component;
 
-template <typename T> class ComponentListIterator;
+template <typename T> class OSIMCOMMON_API ComponentListIterator;
```
Comment (chrisdembia) [inline OpenSim/Common/ComponentList.h]: Is this forward declaration supposed to have the `OSIMCOMMON_API` macro?

**Comment by jenhicks** [comment]
@chrisdembia Is this PR still valid?

**Comment by chrisdembia** [comment]
I tried to do two things here; put ComponentList in a separate header and also export it. The exporting doesn't make sense but it would be nice to have the class in a separate header. Closing for now.

**Pull request closed without merging**